testbed/testbed/otelcol_runner.go

```diff
@@ -0,0 +1,170 @@
+	PrepareConfig(configStr string) (string, error)
+	//Starts the otelcol instance(s) if not already running which is the subject of the test to be run.
+	//It returns the host:port of the data receiver to post test data to.
+	Start(args StartParams) (string, error)
+	//Stops the otelcol instance(s) which are the subject of the test just run if applicable. Returns whether
+	//the instance was actually stopped or not.
+	Stop() (bool, error)
+	//WatchResourceConsumption toggles on the monitoring of resource consumpution by the otelcol instance under test.
+	WatchResourceConsumption() error
+	//GetProcessMon returns the Process being used to monitor resource consumption.
+	GetProcessMon() *process.Process
+	//GetTotalConsumption returns the data collected by the process monitor.
+	GetTotalConsumption() *ResourceConsumption
+	//GetResourceConsumption returns the data collected by the process monitor as a display string.
+	GetResourceConsumption() string
+}
+
+//InProcessPipeline implements the OtelcolRunner interfaces running a single otelcol as a go routine within the
+//same process as the test executor.
+type InProcessPipeline struct {
+	logger    *zap.Logger
+	factories config.Factories
+	config    *configmodels.Config
+	svc       *service.Application
+	appDone   chan struct{}
+	stopped   bool
+}
+
+//NewInProcessPipeline crewtes a new InProcessPipeline using the supplied component factories.
+func NewInProcessPipeline(factories config.Factories) *InProcessPipeline {
+	return &InProcessPipeline{
+		factories: factories,
+	}
+}
+
+func (ipp *InProcessPipeline) PrepareConfig(configStr string) (string, error) {
+	logger, err := configureLogger()
+	if err != nil {
+		return "", err
+	}
+	ipp.logger = logger
+	v := viper.NewWithOptions(viper.KeyDelimiter("::"))
+	v.SetConfigType("yaml")
+	v.ReadConfig(strings.NewReader(configStr))
+	cfg, err := config.Load(v, ipp.factories)
+	if err != nil {
+		return "", err
+	}
+	err = config.ValidateConfig(cfg, zap.NewNop())
+	if err != nil {
+		return "", err
+	}
+	ipp.config = cfg
+	return "", nil
+}
+
+func (ipp *InProcessPipeline) Start(args StartParams) (string, error) {
+	params := service.Parameters{
+		ApplicationStartInfo: service.ApplicationStartInfo{
+			ExeName:  "otelcol",
+			LongName: "InProcess Collector",
+			Version:  version.Version,
+			GitHash:  version.GitHash,
+		},
+		ConfigFactory: func(v *viper.Viper, factories config.Factories) (*configmodels.Config, error) {
+			return ipp.config, nil
+		},
+		Factories: ipp.factories,
+	}
+	var err error
+	ipp.svc, err = service.New(params)
+	if err != nil {
+		return "", err
+	}
+	ipp.svc.Command().SetArgs(args.cmdArgs)
+
+	ipp.appDone = make(chan struct{})
+	go func() {
+		defer close(ipp.appDone)
+		appErr := ipp.svc.Start()
+		if appErr != nil {
+			err = appErr
+		}
+	}()
+
+	for state := range ipp.svc.GetStateChannel() {
+		switch state {
+		case service.Starting:
+			// NoOp
+		case service.Running:
+			return DefaultHost, err
```

Comment:
Fixed to include port. For the two existing implementations of OtelcolRunner this value can be safely ignored because it will always be localhost. In the remote k8s implementation it will be important because the value may vary from one test execution to the next. The will be a little extra work that needs to be done yet to pass this returned value into the LoadGenerator so it knows where to post data to.